framework/include/meshlet_helpers.hpp

```diff
@@ -0,0 +1,263 @@
+﻿#pragma once
+#include <gvk.hpp>
+
+namespace gvk
+{
+	/** Meshlet struct for the CPU side. */
+	struct meshlet
+	{
+		/** The model of the meshlet. */
+		model mModel;
+		/** The optional mesh index of the meshlet.
+		 *  Only set if the submeshes were not combined upon creation of the meshlet.
+		 */
+		std::optional<mesh_index_t> mMeshIndex;
+		/** Contains indices into the original vertex attributes of the mesh.
+		 *  If submeshes were combined, it indexes the vertex attributes of the combined meshes as done with get_vertices_and_indices.
+		 */
+		std::vector<uint32_t> mVertices;
+		/** Contains indices into the mVertices vector. */
+		std::vector<uint8_t> mIndices;
+		/** The actual number of vertices inside of mVertices; */
+		uint32_t mVertexCount;
+		/** The actual number of indices in mIndices; */
+		uint32_t mIndexCount;
+	};
+
+	// ToDo: Can we make vertex and index size configurable easily?
```

Comment:
Having the sizes as statics in this case adds just some syntactic sugar. Because then, one can make use of automatic type deduction for siutations as follows:

``` 
template<typename T>
auto get_number_of_vertices(const T& meshlet) {
    return meshlet.sNumVertices;
}
```

The static members do not hurt! They do not cost any memory, and potentially make some situations a bit more convenient. And that's already all they are good for.